cloudflare/resource_cloudflare_load_balancer_monitor.go

```diff
@@ -210,14 +246,54 @@ func resourceCloudflareLoadBalancerPoolMonitorUpdate(d *schema.ResourceData, met
 		loadBalancerMonitor.Port = uint16(port.(int))
 	}
 
-	if allowInsecure, ok := d.GetOk("allow_insecure"); ok {
-		loadBalancerMonitor.AllowInsecure = allowInsecure.(bool)
-	}
+	switch loadBalancerMonitor.Type {
+	case "tcp":
+		if method, ok := d.GetOk("method"); ok {
+			loadBalancerMonitor.Method = method.(string)
+		} else {
+			loadBalancerMonitor.Method = "connection_established"
+		}
+	case "http", "https":
+		if allowInsecure, ok := d.GetOk("allow_insecure"); ok {
+			loadBalancerMonitor.AllowInsecure = allowInsecure.(bool)
+		}
+
+		if expectedBody, ok := d.GetOk("expected_body"); ok {
+			loadBalancerMonitor.ExpectedBody = expectedBody.(string)
+		} else {
+			return fmt.Errorf("expected_body must be set")
+		}
+
+		if expectedCodes, ok := d.GetOk("expected_codes"); ok {
+			loadBalancerMonitor.ExpectedCodes = expectedCodes.(string)
+		} else {
+			return fmt.Errorf("expected_codes must be set")
+		}
+
+		if header, ok := d.GetOk("header"); ok {
+			loadBalancerMonitor.Header = expandLoadBalancerMonitorHeader(header)
+		}
+
+		if followRedirects, ok := d.GetOk("follow_redirects"); ok {
+			loadBalancerMonitor.FollowRedirects = followRedirects.(bool)
+		}
+
+		if method, ok := d.GetOk("method"); ok {
+			loadBalancerMonitor.Method = method.(string)
+		} else {
+			loadBalancerMonitor.Method = "GET"
+		}
 
-	if followRedirects, ok := d.GetOk("follow_redirects"); ok {
-		loadBalancerMonitor.FollowRedirects = followRedirects.(bool)
+		if path, ok := d.GetOk("path"); ok {
+			loadBalancerMonitor.Path = path.(string)
+		} else {
+			loadBalancerMonitor.Path = "/"
+		}
 	}
 
+	//
```

Comment:
This can go. The methods are self explanatory.

Reply:
removed